### DISCUSSION TITLE
unbuffered write issue under windows OS, not working.

### DISCUSSION THREAD

**Discussion by wtfiwinomgs** [Chia-Network/bladebit#158] [Q&A]
will there soon be an updated bladebit version that doesn't do unbuffered writes? 

also does bladebit take advantage of avx512 over avx2?
if so, which phase?

**Comment by harold-b** [comment]
> will there soon be an updated bladebit version that doesn't do unbuffered writes?

It's planned, but after some current effort of another kind is completed.

> also does bladebit take advantage of avx512 over avx2?
if so, which phase?

SIMD instructions are only currently used by the "fx" portion in Phase 1 when hashing to Blake3 which has several SIMD implementations. The longest vector one is used whenever the host processor supports it.

I wrote a prototype SIMD portion in Phase 3 (line point creation) which although did get some gains, conversion is already pretty fast that it would likely not make much of a difference but I might re-visit it later.

**Comment by wtfiwinomgs** [comment]
wow awesome and thank you. good to know another version is coming and can finally get it working under windows for unbuffered writes. and avx512 support so some hardware aren't exactly overspent.
